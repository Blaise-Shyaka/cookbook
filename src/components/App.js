import Nav from './Nav';
import Meals from './Meals';

import {
  Switch,
  Route,
} from "react-router-dom";
import SingleMeal from './SingleMeal';

function App() {
  return (
    <div className="App">
      <Nav />
      <Switch>
        <Route exact path="/" component={Meals}></Route>
        <Route path="/meal/:id" component={SingleMeal}></Route>
      </Switch>
    </div>
  );
}

export default App;
