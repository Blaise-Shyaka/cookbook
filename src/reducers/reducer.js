import { FILTERBYCATEGORY, FILTERBYAREA, SEARCH, UPDATEMEALS, GETCATEGORIES, GETAREAS, GETONEMEAL } from '../actions/index';
const defaultState = {
  categoryFilter: '',
  areaFilter: '',
  searchTerm: '',
  meals: [],
  categories: [],
  areas: [],
  oneMeal: '',
}

export default function reducer(state=defaultState, action) {
  const { type, payload } = action
  switch(type) {
    case FILTERBYCATEGORY:
      return { ...state, categoryFilter: payload}
    case FILTERBYAREA:
      return { ...state, areaFilter: payload}
    case SEARCH:
      return { ...state, searchTerm: payload}
    case UPDATEMEALS:
      return { ...state, meals: [...payload] }
    case GETCATEGORIES:
      return { ...state, categories: [...payload]}
    case GETAREAS:
      return { ...state, areas: [...payload]}
    case GETONEMEAL:
      return { ...state, oneMeal: payload}
    default:
      return state
  }
}