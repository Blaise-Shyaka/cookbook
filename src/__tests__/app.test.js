import configureStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../components/App';
import { defaultState } from '../reducers/reducer';

const mockStore = configureStore([thunkMiddleware]);
const store = mockStore(defaultState);

describe('Meals Component', () => {
  const appComponent = (
    <Provider store={store}>
      <Router>
        <App />
      </Router>
    </Provider>
  );

  it('Should render correctly', () => {
    render(appComponent);
  });

  it('Should display a list of meal items', () => {
    const tree = renderer.create(appComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
