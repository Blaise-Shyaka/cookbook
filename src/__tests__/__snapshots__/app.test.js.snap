// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meals Component Should display a list of meal items 1`] = `
<div>
  <nav
    className="navWrapper"
  >
    <div
      className="logo"
    >
      <p>
        CookBook
      </p>
    </div>
    <div
      className="searchWrapper"
    >
      <input
        id="searchInput"
        placeholder="Search Meal By Name"
        type="search"
      />
      <button
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
    <div
      className="filterWrapper"
    >
      <select
        defaultValue="Filter By Category"
        onChange={[Function]}
      >
        <option
          disabled={true}
          value="Filter By Category"
        >
          Filter By Category
        </option>
      </select>
      <select
        defaultValue="Filter By Area"
        onChange={[Function]}
      >
        <option
          disabled={true}
          value="Filter By Area"
        >
          Filter By Area
        </option>
      </select>
    </div>
  </nav>
  <div>
    <div />
  </div>
</div>
`;
